provider_run.go

```diff
@@ -424,7 +425,13 @@ func runMigration(ctx context.Context, db database.DBTxConn, m *Migration, direc
 
 // runGo is a helper function that runs the given Go functions in the given direction. It must only
 // be called after the migration has been initialized.
-func runGo(ctx context.Context, db database.DBTxConn, m *Migration, direction bool) error {
+func runGo(ctx context.Context, db database.DBTxConn, m *Migration, direction bool) (retErr error) {
+	defer func() {
+		if r := recover(); r != nil {
+			retErr = fmt.Errorf("panic: %v\n%s", r, debug.Stack())
```

Comment:
The default Go stack is a bit noisy, wonder if there's an opportunity here to clean this up a bit, i.e., pretty print stack.